src/main/java/io/anserini/index/IndexReaderUtils.java

```diff
@@ -198,6 +248,16 @@ public static String analyzeTerm(String term, Analyzer analyzer) {
     return termInfo;
   }
 
+  /**
+   * Returns iterator over all terms in the collection.
+   * @param reader index reader
+   * @return IndexTerm wrapper over TermsEnum
+   * @throws IOException if error encountered during access to index
+   */
+  public static IndexTerm getTermIterator(IndexReader reader) throws IOException {
```

Comment:
Don't we want something like `public static Iterator<Term> getTerms(IndexReader reader)`?
The point is that `TermsEnum` is a weird Lucene construct... if we wrap into a normal Java Iterator, we can do standard Java stream operations like `map` and `forEach`...

Make sense?

Reply:
It does. Sorry for the back and forth, could you clarify this for me? To implement a custom iterator to do what you suggested, I would do something like below, but the `iterator` method isn't supposed to have any parameters as far as I know? How do we get around that?

```
public Iterator<IndexTerm> iterator(IndexReader reader) {
  return new IndexTermIterator(reader);
}

public class IndexTermIterator<IndexTerm> implements Iterator<IndexTerm> {
  TermsEnum curTerm;

  public IndexTermIterator(IndexReader reader) {
    this.curTerm = MultiTerms.getTerms(reader, "contents").iterator();
  }

  @Override
  public boolean hasNext() {
    return this.curTerm != null;
  }

  @Override
  public IndexTerm next() {
    return new IndexTerm(this.curTerm);
  }
}
```

Reply:
Ah, `iterator()` is only necessary if you want to use it something like a `for (String term : X)`.

In this case, `Iterator<Term> getTerms(IndexReader reader)` is fine...

Also, you can use an anonymous inner class instead of defining another class...